tools/c7n_gcp/c7n_gcp/resources/gke.py

```diff
@@ -78,6 +80,57 @@ def augment(self, resources):
         return resources
 
 
+@KubernetesCluster.filter_registry.register('server-config')
+class ClusterServerConfig(ValueFilter):
+    """Filters kubernetes nodepools by their server config.
+    See `getServerConfig
+    https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations/getServerConfig
+    for valid fields.
+
+    :example:
+
+    Filter all instances that have a firewall rule that allows public
+    acess
+
+    .. code-block:: yaml
+
+        policies:
+           - name: find-unsupported-cluster-version
+             resource: gcp.gke-cluster
+             filters:
+             - type: server-config
+               key: contains(validNodeVersions,currentMasterVersion)
+               value: false
+    """
+
+    schema = type_schema('server-config', rinherit=ValueFilter.schema)
+    permissions = ('projects.locations.getServerConfig',)
+
+    def process_resource(self, client, project, resource):
+        location = resource['location']
+        response = client.execute_command(
+            'getServerConfig', verb_arguments={
+                    'name': 'projects/{}/locations/{}'.format(project, location)}
+        )
+        response['currentMasterVersion'] = resource['currentMasterVersion']
```

Comment:
Why the copy of current value to the response? Ie a standard value filter would cover this outside of the usage of the dedicate config filter.